apps/etl-func/avro/message.json

```diff
@@ -0,0 +1,101 @@
+{
+  "name": "message",
+  "type": "record",
+  "namespace": "dto",
+  "doc": "Kafka JS schema for cosmos api container 'messages'",
+  "fields": [
+    {
+      "name": "schema_version",
+      "doc": "An integer that indicates the version of the schema.",
+      "type": "int"
+    },
+    {
+      "name": "op",
+      "type": {
+        "type": "enum",
+        "doc": "List of operations that can be done to a message. Messages are immutable so we can only perform create operations.",
+        "name": "MessageCrudOperation",
+        "symbols": ["CREATE"]
+      }
+    },
+    {
+      "name": "id",
+      "doc": "An ULID that identifies the message inside the message collection.",
+      "type": "string"
+    },
+    {
+      "name": "sender_service_id",
+      "doc": "The id of the service that sent the message.",
+      "type": "string"
+    },
+    {
+      "name": "sender_user_id",
+      "doc": "The id of the user that sent the message.",
+      "type": "string"
+    },
+    {
+      "name": "is_pending",
+      "doc": "A boolean that indicates if the message has been processed, is_pending equals false indicates that the message has not been processed yet.",
+      "type": "boolean"
+    },
+    {
+      "name": "content_subject",
+      "doc": "The subject of the message.",
+      "type": "string"
+    },
+    {
+      "name": "content_type",
+      "doc": "Indicates the type of the message.",
+      "type": [
+        "null",
+        {
+          "type": "enum",
+          "name": "MessageContentType",
+          "symbols": ["GENERIC", "PAYMENT", "EU_COVID_CERT", "LEGAL", "PN"]
```

Comment:
- Can we remove `EU_COVID_CERT` and `LEGAL`? Are they still used? 
- `PN` and `PAGOPA_RECEIPT` are not really message type, why don't we split this field into 2 and move these values into another field?

Reply:
Maybe we can change the name of this field - its purpose is to "group" messages based on its content
It is not an actual "message type"

We have to review all the categories to make sure that they are correct and actually used

Reply:
I don't think that we can remove any of the symbols here even if they are no longer used cause we are going to process all the messages.
I agree that "message type" is not a proper term here, we can rename this field to something like "message-category" @lucacavallaro @acuiuli

Reply:
You are right about removing values.

I still think that splitting this into 2 fields is the better choice. What happens when `PN` sends a payment message? What values do you choose?

Reply:
@acuiuli In that case, the value should remain 'PN'. The purpose of this field is to map a message to a 'business/initiative category.' We want to track how many messages are sent by PN, how many are pagoPA receipts, etc.

Reply:
I thought that we also want to track how many are generic messages and how many contains a payment, with only one field we will never be able to do that accurately because a payment message sent by PN will not be counted as a payment message.
I think that we need to know if what I'm saying is a requirement or we can just ignore it and let this field as it is and just modify it's name.

Reply:
> I thought that we also want to track how many are generic messages and how many contains a payment

To count how many contains a payment we can check the other (payment related) fields or maybe we can add another field called `has_payment`. 

Our only requirement right now is to include a "business category" to distinguish between different initiatives.

Reply:
So do we agree on renaming this to `message-category`? Do you think it's a good compromise? @acuiuli @lucacavallaro

Reply:
I think we should involve the rest of the team tomorrow morning to decide both the name (`category` is ok?) of the field and its content.